### ISSUE TITLE
Add rule name to custom messages

### ISSUE THREAD

**Issue by jhae-de** [stylelint/stylelint#8773]
There is currently an inconsistency in how error messages are output by rules. Especially with custom messages, the rule name is not included in the error output if it is not explicitly specified. This also applies, for example, to the [stylelint-config-standard](https://github.com/stylelint/stylelint-config-standard) configuration and its `pattern` rules.

### Current behavior
Custom messages only contain the rule name if it is manually added.

### Expected Behavior
All error messages, including custom ones, should always include the rule name for better traceability.

### Example

**Stylelint configuration**
```javascript
{
  'custom-property-pattern': [
    '^([a-z][a-z0-9]*)(-[a-z0-9]+)*$',
    {
      message: (name) => `Expected custom property name "${name}" to be kebab-case`,
    },
  ],
}
```

**CSS code**
```css
:root {
  --MyCustomProperty: '';
}
```

**Current output**
```
Expected custom property name "--MyCustomProperty" to be kebab-case
```

**Expected output**
```
Expected custom property name "--MyCustomProperty" to be kebab-case (custom-property-pattern)
```
